tests/integration/mounts.bats

```diff
@@ -211,6 +211,29 @@ function test_mount_order() {
 	[[ "$output" == *"must be mounted on ordinary directory"* ]]
 }
 
+# https://github.com/opencontainers/runc/issues/4401
+@test "runc run [setgid / + mkdirall]" {
+	mkdir rootfs/setgid
+	chmod '=7755' rootfs/setgid
+
+	update_config '.mounts += [{
+		type: "tmpfs",
+		source: "tmpfs",
+		destination: "/setgid/a/b/c",
+		options: ["ro", "nodev", "nosuid"]
+	}]'
+	update_config '.process.args |= ["true"]'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+
+	# Verify that the setgid bit is inherited.
```

Comment:
Just curious: this works just because, even if mkdir will ignore those bits, linux will honor it when doing the mkdir, as it is inside a dir with setgid?

Reply:
The ignoring and inheriting behaviours are two separate issues. This test works because `S_ISGID` on a directory will cause any new directory created inside it to also have `S_ISGID` set.

Reply:
Cool, that is exactly what I was saying :)